core/src/main/scala/kafka/log/Log.scala

```diff
@@ -787,11 +787,13 @@ class Log(@volatile var dir: File,
    *
    * @param records The records to append
    * @param isFromClient Whether or not this append is from a producer
+   * @param apiVersion Inter-broker message protocol version
    * @throws KafkaStorageException If the append fails due to an I/O error.
    * @return Information about the appended messages including the first and last offset.
    */
-  def appendAsLeader(records: MemoryRecords, leaderEpoch: Int, isFromClient: Boolean = true): LogAppendInfo = {
-    append(records, isFromClient, assignOffsets = true, leaderEpoch)
+  def appendAsLeader(records: MemoryRecords, leaderEpoch: Int, isFromClient: Boolean = true,
+                     apiVersion: ApiVersion = ApiVersion.latestVersion): LogAppendInfo = {
```

Comment:
Maybe we could name this explicitly as `interBrokerProtocolVersion`? Same for the others.